sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -152,6 +153,18 @@ case class Sort(
     global: Boolean,
     child: LogicalPlan) extends UnaryNode {
   override def output = child.output
+
+  override def resolveChildren(name: String, resolver: Resolver) = {
```

Comment:
This feels kind of hacky to me as its mixing the particulars of analysis for SQL into the logical plan.  Could we instead just make resolve not do partial resolution, where we can resolve the base attribute but not the GetFields that are on top.  I think this change is the root cause of the regression.